transport_plugins/bled112/iotile_transport_bled112/virtual_bled112.py

```diff
@@ -182,35 +229,80 @@ def _queue_reports(self, *reports):
+                if self.device:
+                    if self.virtual_info['advertising_version'] == 1:
+                        self._defer(self._command_task.sync_command, [['_set_advertising_data', 1, self._scan_response()]])
+                    else:
+                        self._defer(self._command_task.sync_command, [['_set_advertising_data', 0, self._advertisement()]])
+
                 continue
 
             self.reports.put(report)
 
     def _advertisement(self):
-        # Flags are
-        # bit 0: whether we have pending data
-        # bit 1: whether we are in a low voltage state
-        # bit 2: whether another user is connected
-        # bit 3: whether we support robust reports
-        # bit 4: whether we allow fast writes
-
-        flags = (0 << 1) | (0 << 2) | (1 << 3) | (1 << 4) | (int(self.device.pending_data))
+        # Flags for version 1 are:
+        #   bit 0: whether we have pending data
+        #   bit 1: whether we are in a low voltage state
+        #   bit 2: whether another user is connected
+        #   bit 3: whether we support robust reports
+        #   bit 4: whether we allow fast writes
+        # Flags for version 2 are:
+        #   bit 0: User is connected
+        #   bit 1: POD has data to stream
+        #   bit 2: Broadcast data is encrypted and authenticated using AES-128 CCM
+        #   bit 3: Broadcast key is device key (otherwise user key)
+        #   bit 4-7: Reserved
+
         ble_flags = struct.pack("<BBB", 2, 1, 0x4 | 0x2)  # General discoverability and no BR/EDR support
-        uuid_list = struct.pack("<BB16s", 17, 6, TileBusService.bytes_le)
-        manu = struct.pack("<BBHLH", 9, 0xFF, ArchManuID, self.device.iotile_id, flags)
 
-        return ble_flags + uuid_list + manu
+        if self.virtual_info['advertising_version'] == 1:
+            flags = (0 << 1) | (0 << 2) | (1 << 3) | (1 << 4) | (int(self.device.pending_data))
+            uuid_list = struct.pack("<BB16s", 17, 6, TileBusService.bytes_le)
+            manu = struct.pack("<BBHLH", 9, 0xFF, ArchManuID, self.device.iotile_id, flags)
+            return ble_flags + uuid_list + manu
+
+        else: #self.virtual_info['advertising_version'] == 2:
+            flags = (0 << 0) | (int(self.device.pending_data) << 1) | (int(self.device.connected) << 2) | (0 << 3)
+            reboots = self.virtual_info['reboot_count']
+            voltage = int(self.virtual_info['battery_voltage'] * 32)
+
+            OTHER = 0  #TODO
+            subsecond_cnt = 0xF #TODO
+
+            reserved = 0
+            with self._broadcast_lock:
+                if self._broadcast_reading is not None:
+                    bcast_stream = self._broadcast_reading.stream
+                    bcast_value = self._broadcast_reading.value
+                    timestamp = calendar.timegm(time.gmtime())
+                else:
+                    #XXX: OK?
```

Comment:
I would recommend not doing this.  It means that every access point we load this code on will broadcast a 'deadbeef' broadcast value.  If you're looking for a sentinel value to indicate no broadcast value, we support the following:

stream: 0xFFFF
timestamp: 0xFFFFFFFF
value: 0xFFFFFFFF

The only important part is the reserved invalid stream id `0xFFFF` but the v1 advertisement sets everything to FF so might as well be consistent.